docs/architecture/adr-0028-tracing.md

```diff
@@ -0,0 +1,164 @@
+[AWS X-Ray](https://aws.amazon.com/xray/), or [Honeycomb](https://www.honeycomb.io/).
+
+We'd like to assess how difficult it would be to:
+
+- get started with a logging implementation
+- eventually hook into one of these external services
+
+# Decision
+
+We should use `scala-opentracing` for our initial implementation of tracing. There
+are several reasons why this is the case. The general risk in each of these libraries
+is that they're maintained seemingly by a single person. However, in `scala-opentracing`'s
+case, we at least know that the person maintaining it is responsible for the health of
+a production system. It's not super challenging to get started with `scala-opentracing`.
+I was able to get the logging tracer up and running in under two hours. Chris was able
+to make substantive changes to it in order to extract a header from the API under 90
+minutes after he started looking at this ADR. I was later able to configure dumping
+traces to Jaeger instead of just logging without considerable difficulty.
+
+Those are all very different from the experiences with the other two libraries. `natchez`
+is advertised as not quite ready for production, and even "hook it into `http4s`" is a
+hard problem right now. There's some [ongoing work](https://github.com/tpolecat/natchez/issues/5)
+on that front that looks close to done, but it's also been ongoing for two months, and there's
+no timeline. `http4s-tracer` uses a separate vocabulary from what's used in the
+Open Tracing spec and doesn't have an API for writing traces to external services.
+
+# Consequences
+
+We'll need to add `scala-opentracing` to backsplash. This is fairly easy. Getting
+the tracer configured will also be fairly easy as long as we're just doing a logging
+tracer to start. There's some infra work necessary once we want to add third party
+service.
+
+We'll need to choose a repository for our traces. We'd like that to be X-Ray, since
+it means we don't have to configure and deploy another service, but that may not be
+an option. We can also decide that we should just log for a while, since even that
+would be useful. For reference, changes necessary to add Jaeger as a repository
+for local development are in
+[this pr](https://github.com/jisantuc/tracing-demos/pull/1/files).
+
+We'll need to figure out what we want to trace. We could, for example, make all of
+the color correction and mosaic implicits functions take implicit trace contexts.
+I believe it's the case that if we had, e.g., Jaeger as the repository, we could
+set up the sampler to only send traces on one percent of requests, which would mean
+we could deploy some really aggressive tracing and pay whatever price we pay for
+tracing on one tile in every three or so map loads. If that's true, we should trace
+aggressively. If it's not, we should trace conservatively.
+
+# The Contenders
```

Comment:
I think this section should be moved up to either context or decision -- the descriptions here are important for understanding the decision I think and the ADR reads more naturally with the examination up front in my opinion.

Reply:
Moved, decreased headers one level